src/libraries/Client/Microsoft.Agents.Connector/RestClients/UserTokenRestClient.cs

```diff
@@ -250,5 +250,42 @@ public async Task<TokenResponse> ExchangeTokenAsync(string userId, string connec
                     throw new HttpRequestException($"ExchangeTokenAsync {httpResponse.StatusCode}");
             }
         }
+
+        internal HttpRequestMessage CreateGetTokenOrSignInResourceRequest(string userId, string connectionName, string channelId, string code, string state, string finalRedirect, string fwdUrl)
+        {
+            var request = new HttpRequestMessage();
+            request.Method = HttpMethod.Get;
+
+            request.RequestUri = new Uri(_transport.Endpoint, "api/usertoken/GetTokenOrSignInResource")
+                .AppendQuery("userId", userId)
+                .AppendQuery("connectionName", connectionName)
+                .AppendQuery("channelId", channelId)
+                .AppendQuery("code", code)
+                .AppendQuery("state", state)
+                .AppendQuery("finalRedirect", finalRedirect)
+                .AppendQuery("fwdUrl", fwdUrl);
+
+            request.Headers.Add("Accept", "application/json");
+            return request;
+        }
+
+        /// <inheritdoc/>
+        public async Task<TokenOrSignInResourceResponse> GetTokenOrSignInResourceAsync(string userId, string connectionName, string channelId = null, string code = null, string state = null, string finalRedirect = null, string fwdUrl = null, CancellationToken cancellationToken = default)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(userId);
+            ArgumentException.ThrowIfNullOrEmpty(connectionName);
+            ArgumentException.ThrowIfNullOrEmpty(channelId);
```

Comment:
Is nullcheck expected since in method definition channelId seems an optional param `string channelId = null`